client.go

```diff
@@ -59,19 +59,30 @@ func NewClientWithTimeout(timeout time.Duration) *Client {
 }
 
 type federationTripper struct {
-	transport http.RoundTripper
+	transports map[string]http.RoundTripper
 }
 
 func newFederationTripper() *federationTripper {
-	// TODO: Verify ceritificates
-	return &federationTripper{
-		transport: &http.Transport{
+	tripper := new(federationTripper)
+	tripper.transports = make(map[string]http.RoundTripper)
+	return tripper
+}
+
+// getTransport returns a http.Transport instance with a TLS configuration using
+// the given server name for SNI. It also creates the instance if there isn't
+// any for this server name.
+func (f *federationTripper) getTransport(tlsServerName string) http.RoundTripper {
+	if transport, ok := f.transports[tlsServerName]; !ok || transport == nil {
```

Comment:
could you add a comment here explaining why we need to do this rather than reuse the same transport?